http/headers/accept-ch.json

```diff
@@ -93,6 +93,342 @@
             "status": {
               "experimental": true,
               "standard_track": false,
+              "deprecated": true
+            }
+          }
+        },
+        "Sec-CH-UA": {
```

Comment:
@hamishwillee: Help me understand something here. Servers include `Accept-CH` headers in responses; clients are supposed to react to those headers by setting corresponding headers in their next request. A server sends `Accept-CH: Sec-CH-UA` and a client is supposed set `Sec-CH-UA: somevalue`.

My question is this: what is this feature representing, exactly? That a client sends `Sec-CH-UA` headers? That a client sends `Sec-CH-UA` in response to `Accept-CH: Sec-CH-UA`? Or something else to do with clients processing `Accept-CH: Sec-CH-UA` responses?

I think the answer to that question will suggest something about whether this feature should be a child of `Accept-CH` or ought to be its own entry as a request header alone.

Reply:
@ddbeck Sorry for delay in response! 

> Help me understand something here. Servers include `Accept-CH` headers in responses; clients are supposed to react to those headers by setting corresponding headers in their next request. A server sends `Accept-CH: Sec-CH-UA` and a client is supposed set `Sec-CH-UA: somevalue`.

Yes, with very minor caveats:
- A server specifies  `Accept-CH` with a header in any/every response to tell clients what hints it is capable of using, if provided. 
- Clients may opt-in to responding with these hints (i.e. it is optional, and defined by the client). So more "could" than "is supposed to".
- The client can respond if and when it wants, not just the "next request". Could be the next request, or it could be every subsequent request where it makes sense to do so. 

> My question is this: what is this feature representing, exactly? That a client sends `Sec-CH-UA` headers? That a client sends `Sec-CH-UA` in response to `Accept-CH: Sec-CH-UA`? Or something else to do with clients processing `Accept-CH: Sec-CH-UA` responses?

Good question. For Accept-CH I think it just means client "understands" this directive and may send the header indicated in the directive as a response.

> I think the answer to that question will suggest something about whether this feature should be a child of `Accept-CH` or ought to be its own entry as a request header alone.

Currently it is both. 
-  I did not invent this approach - it just extends what was done before.
- It is useful to have both because the docs for Accept-CH then show the versions in which you can add the various directives. 
- Just FYI, Accept-CH is in a different spec from the actual headers it will get in responses (i.e. Accept-CH is in the client hints infrastructure spec, but there are numerous different client hints that live in their own specs.

Hope that clarifies. Happy to be lead by you!

Reply:
Can I please get a review on this?

Reply:
> > My question is this: what is this feature representing, exactly? That a client sends `Sec-CH-UA` headers? That a client sends `Sec-CH-UA` in response to `Accept-CH: Sec-CH-UA`? Or something else to do with clients processing `Accept-CH: Sec-CH-UA` responses?
> 
> Good question. For Accept-CH I think it just means client "understands" this directive and may send the header indicated in the directive as a response.

Sorry, I have to poke at this a little more (because I don't get it—not as a critique). On the `Accept-CH` side, what does "understand" mean? How would I test this to know if it's supported or not, since there's nuance about some-or-all subsequent requests? (I saw the link to the demo, but the view source link doesn't work and I can't figure out what's supposed to happen with just the browser dev tools.)

Perhaps more critically, when would the compat data for `http.headers.Accept-CH.Sec-CH-UA` be different from the compat data for `http.headers.Sec-CH-UA`?

> > I think the answer to that question will suggest something about whether this feature should be a child of `Accept-CH` or ought to be its own entry as a request header alone.
>
> …
>
> * I did not invent this approach - it just extends what was done before.

Can you point to an example of this and maybe provide a little commentary on it? I feel a bit dumb here, but I can't figure out what this structure is similar to.

> * Just FYI, Accept-CH is in a different spec from the actual headers it will get in responses (i.e. Accept-CH is in the client hints infrastructure spec, but there are numerous different client hints that live in their own specs.

That's fine. That's why we can put a spec URL on any feature. 😆